### PR TITLE
moves schedule_tasks... out of cycle

### PR DESCRIPTION
**Pull request by JamesTFarrington**
also creates a max length for event queue.

Reorganizes trabajo so that methods called by schedule_tasks_for_queueing are all together and in order.

### CODE REVIEW

**Comment by vepkenez** [comment]
This looks like an improvement.  I am still wondering though, _why_ do we schedule so many future tasks?  Why not just say "run this now"?  Why do we say "go get all future runnings of this task and schedule them all"?

**Comment by JamesTFarrington** [comment]
This behavior only happens for tasks where force_interval is specified

File: tiempo/tests/test_scheduling.py [hunk 1]
```diff
@@ -112,11 +112,13 @@ def test_dependent_task_is_queued_after_trigger_task(self):
 
     def test_unscheduled_task_is_unplanned(self):
         unscheduled_task = unplanned_task(some_callable)
-        self.assertFalse(unscheduled_task.is_planned())
+        boolean = unscheduled_task.force_interval or unscheduled_task.periodic
```
Comment (jMyles) [inline tiempo/tests/test_scheduling.py]: Why this change?  We aren't testing the `is_planned` method now.

Comment (JamesTFarrington) [inline tiempo/tests/test_scheduling.py]: I've deleted is_planned
